bioptim/dynamics/configure_problem.py

```diff
@@ -612,9 +747,36 @@ def configure_taudot(nlp, as_states: bool, as_controls: bool):
             If the generalized force derivatives should be a control
         """
 
-        name_tau = [str(i) for i in range(nlp.model.nbGeneralizedTorque())]
+        name_taudot = [str(i) for i in range(nlp.model.nbGeneralizedTorque())]
         ConfigureProblem._adjust_mapping("taudot", ["qdot", "tau"], nlp)
-        ConfigureProblem.configure_new_variable("taudot", name_tau, nlp, as_states, as_controls)
+        ConfigureProblem.configure_new_variable("taudot", name_taudot, nlp, as_states, as_controls)
+
+    @staticmethod
+    def configure_soft_contact_forces(nlp, as_states: bool, as_controls: bool):
+        """
+        Configure the generalized forces derivative
+
+        Parameters
+        ----------
+        nlp: NonLinearProgram
+            A reference to the phase
+        as_states: bool
+            If the generalized force derivatives should be a state
+        as_controls: bool
+            If the generalized force derivatives should be a control
+        """
+        name_soft_contact_forces = []
+        l = ["fx", "fy", "fz"]  # TODO: find a better place to hold this or define it in biorbd ?
+        for ii in range(nlp.model.nbSoftContacts()):
+            name_soft_contact_forces.extend(
+                    [
+                        f"{nlp.model.softContactName(ii).to_string()}_{name}"
+                        for name in l
+                        if nlp.model.softContactName(ii).to_string() not in name_soft_contact_forces
+                    ]
+                )
+
+        ConfigureProblem.configure_new_variable("fext", name_soft_contact_forces, nlp, as_states, as_controls)
```

Comment:
`External forces`

Reply:
not as convenient, and not as compact as fext and/or qdot, q, tau, etc..., it make sens for me to have mathematical notations. When calling nlp.controls["extenal forces"], it doesn't look fine.

Reply:
Oh! I thought it was for the name of the plot
Yep, I agree